pkg/provider/validators.go

```diff
@@ -51,3 +51,17 @@ func validateUUID(v interface{}, _ cty.Path) diag.Diagnostics {
 	}
 	return nil
 }
+
+func validateClusterName(v interface{}, _ cty.Path) diag.Diagnostics {
```

Comment:
I think we don't need to do this validation. 

API will already return a 400 - Bad Request, when there is no name provided for the biganimal_cluster data-source. 
![Screenshot 2023-03-24 at 08 57 41](https://user-images.githubusercontent.com/121254/227459417-77a14d94-e910-477b-b776-c8abfbde93f2.png)
WDYT, @edbamolvyavahare ?

Reply:
If the "name" argument is validated during the input phase, it would be beneficial to promptly alert users of any errors.